www/src/components/email-capture-form.js

```diff
@@ -0,0 +1,160 @@
+import React from "react"
+import { rhythm } from "../utils/typography"
+import presets from "../utils/presets"
+import jsonp from "jsonp"
+
+// Mailchimp endpoint
+// From: https://us17.admin.mailchimp.com/lists/integration/embeddedcode?id=XXXXXX
+// Where `XXXXXX` is the MC list ID
+// Note: we change `/post` to `/post-json`
+const MAILCHIMP_URL = `https://gatsbyjs.us17.list-manage.com/subscribe/post-json?u=1dc33f19eb115f7ebe4afe5ee&amp;id=f366064ba7`
+
+class EmailCaptureForm extends React.Component {
+  constructor() {
+    super()
+    this.state = {
+      email: ``,
+    }
+  }
+
+  // Update state each time user edits their email address
+  _handleEmailChange = e => {
+    this.setState({ email: e.target.value })
+  }
+
+  // Check whether the email address is valid:
+  // - not an empty string,
+  // - greater than 5 characters,
+  // - includes both `@` and `.`
+  _isValidEmailAddress = email =>
```

Comment:
It's better to off-load common algorithms like this to an NPM module e.g. https://github.com/Sembiance/email-validator